src/realizations/catchment/Bmi_Module_Formulation.cpp

```diff
@@ -35,13 +34,40 @@ namespace realization {
             }
 
             std::string output_str;
-            for (const std::string& name : get_output_variable_names()) {
-                output_str += (output_str.empty() ? "" : ",") + std::to_string(get_var_value_as_double(0, name));
+            for (int i = 0; i < get_output_variable_names().size(); ++i) {
+                std::string name = get_output_variable_names()[i];
+                std::string output_units = get_output_variable_units()[i];
+                std::string out_units_norm = (output_units.empty() || output_units == "none") ? "1" : output_units;
+                double var_value;
+                try{
+                    var_value = get_value(CatchmentAggrDataSelector(this->get_catchment_id(), name, 0, 0, output_units), MEAN);
+                }
+                catch(data_access::unit_conversion_exception &uce){
+                    data_access::unit_error_log_key key{get_id(), name, uce.provider_model_name, uce.provider_bmi_var_name, uce.what()};
```

Comment:
```suggestion
                    data_access::unit_error_log_key key{"File output", name, uce.provider_model_name, uce.provider_bmi_var_name, uce.what()};
```
We don't really want per-catchment logs of the same failure. Let's just name it as "File output", so it's clear it's something configured explicitly in the realization configuration file.